lib/route-recognizer.ts

```diff
@@ -178,6 +178,10 @@ function parse(segments: Segment[], route: string, types: [number, number, numbe
     if (flags & SegmentFlags.Named) {
       part = part.slice(1);
       names = names || [];
+      if (part === '') {
+        throw new Error(`Missing name for dynamic segment in '/${route}', please provide a name (e.g. \`path: '${type === SegmentType.Star ? '*' : ':'}nameHere'\`)`);
```

Comment:
We don't have a mechanism to thread our prod/dev build logic into here at the moment.  I'm happy to add that to the list of things to address across our microlibs (we desperately need it), but I'd like to keep that decoupled if possible...